pravega/pravega
# Issue 3826: Cherry-pick commits from master to r0.5

**Change log description**  
* Cherry-picks recent commits fixing bugs from master to r0.5.

**Purpose of the change**  
Fixes #3826

**What the code does**  
Cherry-picks the following commits:

```
Issue 3823: Fix incorrect segment name in exception thrown. (#3824)
Issue 3816: Sporadic failure of BookieFailoverTest in Docker Swarm system tests (#3818)
Issue 3811: Fix SpotBugs error on Java 11 build (#3812)
Issue 3724: (SegmentStore) BugFix - Creating an already existing Segment in Tier 2 (#3804)
Issue 3463: Ensure completed transactions are not pinged by the client. (#3767)
Issue 3805: (SegmentStore) BugFix - StorageWriter not shutting down if Tier 2 fencing detected. (#3806)
```

**How to verify it**  
Check the git log to make sure it contains the expected set of commits and make sure that the Travis build succeeds.
